frontend/packages/local-storage-operator-plugin/src/components/local-volume-discovery/request.ts

```diff
@@ -0,0 +1,83 @@
+import * as _ from 'lodash';
+import {
+  apiVersionForModel,
+  K8sResourceCommon,
+  MatchExpression,
+  Toleration,
+  k8sCreate,
+  k8sPatch,
+  k8sGet,
+} from '@console/internal/module/k8s';
+import { LocalVolumeDiscovery } from '../../models';
+import { DISCOVERY_CR_NAME, HOSTNAME_LABEL_KEY, LABEL_OPERATOR } from '../../constants';
+import { getNodeSelectorTermsIndices } from '@console/local-storage-operator-plugin/src/utils';
+
+export const getDiscoveryRequestData = (
+  nodes: string[],
+  ns: string,
+  toleration?: Toleration,
+): LocalVolumeDiscoveryKind => {
+  const request: LocalVolumeDiscoveryKind = {
+    apiVersion: apiVersionForModel(LocalVolumeDiscovery),
+    kind: LocalVolumeDiscovery.kind,
+    metadata: { name: DISCOVERY_CR_NAME, namespace: ns },
+    spec: {
+      nodeSelector: {
+        nodeSelectorTerms: [
+          {
+            matchExpressions: [
+              {
+                key: HOSTNAME_LABEL_KEY,
+                operator: LABEL_OPERATOR,
+                values: nodes,
+              },
+            ],
+          },
+        ],
+      },
+    },
+  };
+  if (!_.isEmpty(toleration)) request.spec.tolerations = [toleration];
+  return request;
+};
```

Comment:
```suggestion
): LocalVolumeDiscoveryKind => ({
    apiVersion: apiVersionForModel(LocalVolumeDiscovery),
    kind: LocalVolumeDiscovery.kind,
    metadata: { name: DISCOVERY_CR_NAME, namespace: ns },
    spec:{
      nodeSelector: {
        nodeSelectorTerms: [
          {
            matchExpressions: [
              {
                key: HOSTNAME_LABEL_KEY,
                operator: LABEL_OPERATOR,
                values: nodes,
              },
            ],
          },
        ],
      },
    ...(!_.isEmpty(tolerations) ? {tolerations} : {})
  };
```

Reply:
I think the current is fine as well.
Not sure why we need to destructure an empty object and not use if.